shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1978,7 +1978,8 @@ void WebContents::LoadURL(const GURL& url,
   // Calling LoadURLWithParams() can trigger JS which destroys |this|.
   auto weak_this = GetWeakPtr();
 
-  params.transition_type = ui::PAGE_TRANSITION_TYPED;
+  params.transition_type = ui::PageTransitionFromInt(
+      ui::PAGE_TRANSITION_TYPED | ui::PAGE_TRANSITION_FROM_API);
```

Comment:
What is the reasoning for adding the transition type `ui::PAGE_TRANSITION_FROM_API` ?

Reply:
It probably is sufficient, but FROM_API seemed more semantically correct. The docs say FROM_API is reserved for embedders but it does seem to have some meaning to content:

https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/navigation_controller_impl.cc;l=179;drc=4f5373794ada7eba6da97bff0229fb0d85f601ec

the default impl of `IsExplicitNavigation` checks for `ui::PAGE_TRANSITION_FROM_ADDRESS_BAR`. I'm not sure whether it's correct to include that here.

Reply:
Browser navigations code paths from `headless`, `shell` components seem to use `ui::PAGE_TRANSITION_FROM_ADDRESS_BAR` along with the typed core transition type, so it should be safe to use here as well.

Reply:
Alrighty, I'll switch this to use `ui::PAGE_TRANSITION_TYPED | ui::PAGE_TRANSITION_FROM_ADDRESS_BAR`.